dnsimple/dnsimple.go

```diff
@@ -342,11 +342,37 @@ func CheckResponse(resp *http.Response) error {
 		return nil
 	}
 
+	bodyBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+	resp.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+
 	errorResponse := &ErrorResponse{}
 	errorResponse.HTTPResponse = resp
+	err = json.NewDecoder(resp.Body).Decode(errorResponse)
 
-	err := json.NewDecoder(resp.Body).Decode(errorResponse)
+	// Handle the case where the errors field is a map of strings
 	if err != nil {
+		var typeErr *json.UnmarshalTypeError
+		if errors.As(err, &typeErr) && typeErr.Field == "errors" {
+			resp.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+			
+			var alternateResp struct {
+				Message string `json:"message"`
+				AttributeErrors  map[string]string `json:"errors"`
+			}
```

Comment:
Have you considered defining the alternate response structure at package level rather than inline?

Reply:
That makes sense to me; I made the change [here](https://github.com/dnsimple/dnsimple-go/pull/191/commits/8625cc849a75b63b13884e0188be310dbc62d125)